docs/deployment.md

```diff
@@ -13,52 +13,65 @@ There are orchestration templates for running Jaeger with:
+primary supported storage backends. 
+=======
+Collectors require a persistent storage backend. Cassandra 3.x (default) and ElasticSearch are the
+primary supported storage backends. There is ongoing work to add support for MySQL and ScyllaDB.
 
-Port | Protocol | Function
----- | -------  | ---
-5775 | UDP      | accept zipkin.thrift over compact thrift protocol
-6831 | UDP      | accept jaeger.thrift over compact thrift protocol
-6832 | UDP      | accept jaeger.thrift over binary thrift protocol
-5778 | HTTP     | serve configs, sampling strategies
+### Cassandra
 
-It can be executed directly on the host or via Docker, as follows:
+A script is provided to initialize Cassandra keyspace and schema
+using Cassandra's interactive shell [`cqlsh`][cqlsh]
 
-```bash
-## make sure to expose only the ports you use in your deployment scenario!
-docker run \
-  --rm \
-  -p5775:5775/udp \
-  -p6831:6831/udp \
-  -p6832:6832/udp \
-  -p5778:5778/tcp \
-  jaegertracing/jaeger-agent
+```sh
+MODE=test sh ./plugin/storage/cassandra/schema/create.sh | cqlsh
 ```
+or if you don't have the source code 
+```
+docker run -e MODE=... jaegertracing/jaeger-cassandra-schema
+=======
+using Cassandra's interactive shell [`cqlsh`][cqlsh],
+clone the source repository following [getting
+started](https://github.com/jaegertracing/jaeger/blob/master/CONTRIBUTING.md#getting-started) and then run:
 
-### Discovery System Integration
+```sh
+MODE=test sh ./plugin/storage/cassandra/schema/create.sh | cqlsh
+```
+For production deployment, pass `MODE=prod DATACENTER={datacenter}` arguments to the script,
+where `{datacenter}` is the name used in the Cassandra configuration / network topology.
 
-The agents can connect point to point to a single collector address, which could be
-load balanced by another infrastructure component (e.g. DNS) across multiple collectors.
-The agent can also be configured with a static list of collector addresses.
+The script also allows overriding TTL, keyspace name, replication factor, etc.
+Run the script without arguments to see the full list of recognized parameters.
 
-On Docker, a command like the following can be used:
+### ElasticSearch
 
-```bash
-docker run \
-  --rm \
-  -p5775:5775/udp \
-  -p6831:6831/udp \
-  -p6832:6832/udp \
-  -p5778:5778/tcp \
-  jaegertracing/jaeger-agent \
-  /go/bin/agent-linux --collector.host-port=jaeger-collector.jaeger-infra.svc:14267
-```
+ElasticSearch does not require initialization other than
+[installing and running ElasticSearch](https://www.elastic.co/downloads/elasticsearch).
+Once it is running, pass the correct configuration values to the Jaeger collector and query service.
 
-In the future we will support different service discovery systems to dynamically load balance
-across several collectors ([issue 213](https://github.com/uber/jaeger/issues/213)).
+#### Shards and Replicas for ElasticSearch indices
+
+=======
```

Comment:
This also looks odd: 

![image](https://user-images.githubusercontent.com/13387/31653408-c03d19de-b322-11e7-96a8-f7375e9ed5b5.png)

Some editors, such as Visual Studio Code, have a markdown viewer that might be useful:
![image](https://user-images.githubusercontent.com/13387/31653534-3051ae60-b323-11e7-9a04-45c5060704f8.png)